osf/management/commands/cumulative_plos_metrics.py

```diff
@@ -0,0 +1,157 @@
+import csv
+import io
+from datetime import datetime
+import logging
+import requests
+import tempfile
+
+from django.core.management.base import BaseCommand
+from keen import KeenClient
+from requests_oauthlib import OAuth2
+
+from addons.osfstorage.models import OsfStorageFile
+from framework.celery_tasks import app as celery_app
+from osf.metrics import PreprintView, PreprintDownload
+from osf.models import Guid, Node, OSFUser, Preprint, \
+    Registration, TrashedFile
+from osf.utils.fields import ensure_str
+from website.settings import KEEN, \
+    PLOS_METRICS_BASE_FOLDER, \
+    PLOS_METRICS_OSF_TOKEN, \
+    PLOS_METRICS_INITIAL_FILE_DOWNLOAD_URL
+
+
+DEFAULT_API_VERSION = '2.14'
+TEMP_FOLDER = tempfile.mkdtemp(suffix='/')
+COL_HEADERS = [
+    'id',
+    'doi',
+    'link',
+    'guid',
+    'type',
+    'views',
+    'downloads',
+]
+
+logger = logging.getLogger(__name__)
+keen_client = KeenClient(
+    project_id=KEEN['public']['project_id'],
+    read_key=KEEN['public']['read_key']
+)
+
+def bearer_token_auth(token):
+    token_dict = {
+        'token_type': 'Bearer',
+        'access_token': token
+    }
+    return OAuth2(token=token_dict)
+
+def upload_metrics_file(file_path, params):
+    with open(file_path, 'rb') as metrics_file:
+        requests.put(
+            url=PLOS_METRICS_BASE_FOLDER,
+            headers={'Accept': 'application/vnd.api+json;version={}'.format(DEFAULT_API_VERSION)},
+            params=params,
+            data=metrics_file,
+            auth=bearer_token_auth(PLOS_METRICS_OSF_TOKEN),
+        )
+
+def parse_base_file():
+    r = requests.get(
+        url=PLOS_METRICS_INITIAL_FILE_DOWNLOAD_URL,
+        headers={'Content-type': 'application/CSV'},
+        auth=bearer_token_auth(PLOS_METRICS_OSF_TOKEN)
+    )
+    csvr = csv.DictReader(ensure_str(r.content).splitlines(), fieldnames=COL_HEADERS)
+    headers = next(csvr)
+    for k, v in headers.items():
+        assert k == v, f'Unexpected header row: {k} and {v} do not match'
```

Comment:
```suggestion
    csvr = csv.DictReader(ensure_str(r.content).splitlines())
    assert csvr.fieldnames == COL_HEADERS, f'Unexpected headers: expected {COL_HEADERS}, got {csvr.fieldnames}'
```

Reply:
I considered this, but order matters, and I wasn't certain this comparison guarantees that.

Reply:
Raw list and/or tuple comparison requires shared ordering. Also confirmed that `csvr.fieldnames` returns a list.
```
>>> [1, 2, 3, 4, 5] == [1, 2, 3, 5, 4]
False
```